<!-- -*- mode: XML -*- -->
<launch>
    <arg name="connect_on_startup"   default="true"/>
    <arg name="calibrate_on_startup" default="true"/>
    <arg name="engage_on_startup"    default="false"/>

    <arg name="odom_frame"        default="odom"/>
    <arg name="base_frame"        default="wheel_odom_link"/>
    <arg name="publish_odom"        default="true"/>
    <arg name="publish_odom_tf"     default="false"/>

    <arg name="publish_current"        default="true"/>
    
    <arg name="output"   default="screen"/>
    <arg name="respawn"   default="false"/>

    <arg name="wheel_track"            default="0.77"/>
    <arg name="tyre_circumference"     default="0.03852"/>
    <arg name ="odrive_sn"             default="3955314C3536"/>
    <arg name ="axis_for_right"        default="1"/>
    <arg name ="flip_left_direction"   default="false"/>
    <arg name ="flip_right_direction"  default="false"/>
    <arg name ="loop_rate"             default="50"/>

    <arg name="torque_control" default="false"/>
    <arg name="velo_p" default="1"/>
    <arg name="velo_i" default="0"/>
    <arg name="velo_d" default="0"/>
    <arg name="turn_p" default="1"/>
    <arg name="turn_i" default="0"/>
    <arg name="turn_d" default="0"/>

    <arg name="odrive_reconfigure"              default="true"/>
    <arg name="controller_vel_gain"             default="1.0"/>
    <arg name="controller_vel_integrator_gain"  default="1.0"/>
    <arg name="encoder_bandwidth"               default="100"/>
    <arg name="motor_current_control_bandwidth" default="100"/>
    <arg name="motor_current_lim"               default="10"/>
    
    <node pkg="differential_odrive" type="odrive_node" name="odrive" ns="odrive" 
            output="$(arg output)" respawn="$(arg respawn)">

        <param name="wheel_track"           value="$(arg wheel_track)"/>
        <param name="tyre_circumference"    value="$(arg tyre_circumference)"/>
        <param name="odrive_sn"             value="$(arg odrive_sn)"/>
        <param name="axis_for_right"        value="$(arg axis_for_right)"/>
        <param name="flip_left_direction"   value="$(arg flip_left_direction)"/>
        <param name="flip_right_direction"  value="$(arg flip_right_direction)"/>
        <param name="loop_rate"             value="$(arg loop_rate)"/>

        <param name="torque_control" value="$(arg torque_control)"/>
        <param name="velo_p" value="$(arg velo_p)"/>
        <param name="velo_i" value="$(arg velo_i)"/>
        <param name="velo_d" value="$(arg velo_d)"/>
        <param name="turn_p" value="$(arg turn_p)"/>
        <param name="turn_i" value="$(arg turn_i)"/>
        <param name="turn_d" value="$(arg turn_d)"/>

        <param name="odrive_reconfigure"              value="$(arg odrive_reconfigure)"/>
        <param name="controller_vel_gain"             value="$(arg controller_vel_gain)"/>
        <param name="controller_vel_integrator_gain"  value="$(arg controller_vel_integrator_gain)"/>
        <param name="encoder_bandwidth"               value="$(arg encoder_bandwidth)"/>
        <param name="motor_current_control_bandwidth" value="$(arg motor_current_control_bandwidth)"/>
        <param name="motor_current_lim"               value="$(arg motor_current_lim)"/>

        <param name="odom_frame"   value="$(arg odom_frame)"/>
        <param name="base_frame"   value="$(arg base_frame)"/>
        
        <param name="connect_on_startup"   value="$(arg connect_on_startup)"/>
        <param name="calibrate_on_startup" value="$(arg calibrate_on_startup)"/>
        <param name="engage_on_startup"    value="$(arg engage_on_startup)"/>
        
        <param name="publish_odom"         value="$(arg publish_odom)"/>
        <param name="publish_odom_tf"      value="$(arg publish_odom_tf)"/>

        <param name="publish_current"      value="$(arg publish_current)"/>
    </node>
</launch>
